Makefile

```diff
@@ -162,19 +162,21 @@ install-lint-tools:
 .PHONY: lint
 lint:
 	golangci-lint --version
-	golangci-lint run --fix --verbose
+	# NOTE: If you get a "Killed" OOM message, try reducing the value of GOGC
+	# See https://github.com/golangci/golangci-lint#memory-usage-of-golangci-lint
+	GOGC=100 golangci-lint run --fix --verbose
 
 .PHONY: build
 build:
 	go build -v `go list ./... | grep -v 'resource_customizations\|test/e2e'`
 
 .PHONY: test
 test:
-	 ./hack/test.sh -coverprofile=coverage.out `go list ./... | grep -v 'test/e2e'`
+	NO_PROXY=* ./hack/test.sh -coverprofile=coverage.out `go list ./... | grep -v 'test/e2e'`
```

Comment:
what does `No_PROXY` do please

Reply:
`NO_PROXY` ensures that the tests surrounding [healthz](https://github.com/argoproj/argo-cd/blob/master/util/healthz/healthz_test.go) don't go out through a proxy server (if one is configured). Golang's `[net/http/transport](https://golang.org/src/net/http/transport.go)` used by `net/http` honors it.

Reply:
ok, please add comment?

Reply:
The alternative is to add code like the following to `healthz_test.go` and `swagger_test.go`:
```go
// ensure that no proxy is being used
// (regardless of how the system is configured via environment variables)
transport := &http.Transport{}
transport.Proxy = nil

client := &http.Client{}
client.Transport = transport

client.Get(...)
```